backend/app/routers/visualization.py

```diff
@@ -0,0 +1,127 @@
+import mimetypes
+
+from beanie import PydanticObjectId
+from fastapi import APIRouter, HTTPException, Depends, Security, Request
+from fastapi.datastructures import UploadFile
+from fastapi.param_functions import File
+from bson import ObjectId
+from minio import Minio
+from typing import Optional, List
+from app import dependencies
+from app.config import settings
+from app.keycloak_auth import get_current_user
+from app.models.files import FileContentType, FileDB
+from app.models.datasets import DatasetDB
+from fastapi import APIRouter, HTTPException, Depends, Security
+from fastapi.security import HTTPAuthorizationCredentials, HTTPBearer
+from app.deps.authorization_deps import FileAuthorization
+
+from app.models.viz_config import VizConfigOut, VizConfigDB, VizConfigIn
+from app.models.metadata import MongoDBRef
+
+router = APIRouter()
+security = HTTPBearer()
+
+
+@router.post("/config", response_model=VizConfigOut)
+async def save_visualization_config(
+    vizconfig_in: VizConfigIn,
+    user=Depends(get_current_user),
+    credentials: HTTPAuthorizationCredentials = Security(security),
+):
+    vizconfig_in = vizconfig_in.dict()
+    # TODO why does it not have right type in the db without the lines below?
+    resource_id = vizconfig_in["resource"]["resource_id"]
+    collection = vizconfig_in["resource"]["collection"]
+    resource_ref = MongoDBRef(collection=collection, resource_id=resource_id)
+    del vizconfig_in["resource"]
+    if collection == "files":
+        file = await FileDB.get(PydanticObjectId(resource_id))
+        if file is not None:
+            viz_config = VizConfigDB(**vizconfig_in, resource=resource_ref)
+            await viz_config.insert()
+            return viz_config.dict()
+        else:
+            raise HTTPException(status_code=404, detail=f"File {resource_id} not found")
+    elif collection == "datasets":
+        dataset = await DatasetDB.get(PydanticObjectId(resource_id))
+        if dataset is not None:
+            viz_config = VizConfigDB(**vizconfig_in, resource=resource_ref)
+            await viz_config.insert()
+            return viz_config.dict()
+        else:
+            raise HTTPException(
+                status_code=404, detail=f"Dataset {resource_id} not found"
+            )
+
+
+@router.get("/{file_id}/config_file", response_model=List[VizConfigOut])
```

Comment:
Can we drop the `config_file` part and maybe combine the three routes below? I don't think we need to know what resource it is attached to in the route. For example `/visualizations/123/config_file`. If you are just checking if the resource exists, since the model has the mongo reference, you could use that to know which collection to query.

Reply:
I now have just one route that checks for these.